src/utils/cache.ts

```diff
@@ -0,0 +1,60 @@
+import { Operator } from "./operators";
+import { Vehicle } from "./interfaces";
+import { logError } from "./logging";
+
+const VEHICLE_CACHE_TTL_IN_MS = 10000;
+
+type VehicleCache = {
+    lastFetched: Date | null,
+    isFetching: boolean,
+    vehicles: Vehicle[] | null
+}
+
+const cache: Record<Operator, VehicleCache> = {
+    [Operator.VOI]: { lastFetched: null, isFetching: false, vehicles: null},
+    [Operator.TIER]: { lastFetched: null, isFetching: false, vehicles: null},
+    [Operator.ZVIPP]: { lastFetched: null, isFetching: false, vehicles: null},
+    [Operator.LIME]: { lastFetched: null, isFetching: false, vehicles: null},
+};
+
+export function getCachedScooters(operator: Operator, fetchVehicles: () => Promise<Vehicle[]>) {
```

Comment:
I staden for `operator` som første  parameter så går det an å ha ein meir generell `key: string`. Då kan du cache Tier med lat, lon, range ved å kalle 
```
getCachedScooters(`tier-${lat}-${lon}-${range}`, () => getTierScooters(lat, lon, range))
```
men å berre droppe caching for Tier er heilt innafor å begynne med.

Reply:
det vil ha ganske dårlig virkning å cache lat-lon med mindre den blir avrundet kraftig på vei inn da. kan også være en memory killer, bokstavelig talt 🔢